src/main/scala/cognite/spark/v1/FlexibleDataModelCorePropertySyncRelation.scala

```diff
@@ -110,54 +110,111 @@ private[spark] class FlexibleDataModelCorePropertySyncRelation(
                 p != "startNode" && p != "endNode" && p != "space" && p != "externalId" && p != "type")
         ))
       .toSeq
+    val cursors = if (cursor.nonEmpty) Some(Map("sync" -> cursor)) else None
+    val select = Map("sync" -> SelectExpression(sources = sourceReference))
+    val (futureItemsCursor, executeBackFill) =
+      generateFutureItemsCursor(cursors, instanceType, select).unsafeRunSync()
+
+    Seq(
+      syncOut(
+        futureItemsCursor,
+        cursors = if (executeBackFill) None else cursors,
+        `with` = Map("sync" -> tableExpression),
+        select,
+        selectedColumns
+      ))
+  }
 
-    val instanceSyncRequest = InstanceSyncRequest(
-      `with` = Map("sync" -> tableExpression),
-      cursors = if (cursor.nonEmpty) Some(Map("sync" -> cursor)) else None,
-      select = Map("sync" -> SelectExpression(sources = sourceReference)),
-      includeTyping = Some(true)
-    )
+  /**
+    * Generate future items cursor and validate cursor expiration.
+    * If the cursor has expired, generate a new cursor and do a full back fill.
+    */
+  private def generateFutureItemsCursor(
+      cursors: Option[Map[String, String]],
+      instanceType: InstanceType,
+      select: Map[String, SelectExpression]): IO[(Option[String], Boolean)] =
+    fetchData(
+      isBackFill = false,
```

Comment:
```suggestion
      useSync = false,
```
Looks like it leaked here that fetchData shouldn't be thinking about backfill or not but rather should just accept whether to use sync or query endpoint